### PR TITLE
Remove Invite and Leave button from Default Room Settings

### PR DESCRIPTION
**Pull request by sobitneupane**
### Details

Remove Invite and Leave button from Default Room Settings
### Fixed Issues

$ https://github.com/Expensify/App/issues/7791

### Tests

1. Open Default Room (#admins and #announce) Settings.
2. Invite and Leave button should be hidden.
3. Open User-Created Room Settings.
4. Invite and Leave button should be visible.

- [x] Verify that no errors appear in the JS console

#### Contributor (PR Author) Checklist
- [x] I verified the PR has a small number of commits behind `main`
- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [ ] I clearly indicated the environment tests should be run in (Staging vs Production)
- [x] I wrote testing steps that cover success & fail scenarios (if applicable)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests & veryfy they passed on:
    - [x] iOS / native
    - [x] Android / native
    - [x] iOS / Safari
    - [x] Android / Chrome
    - [x] MacOS / Chrome
    - [x] MacOS / Desktop
- [x] I verified there are no console errors related to changes in this PR
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I added comments when the code was not self explanatory
    - [ ] I put all copy / text shown in the product in all `src/languages/*` files (if applicable)
    - [ ] I followed proper naming convention for platform-specific files (if applicable)
    - [ ] I followed style guidelines (in [`Styling.md`](https://github.com/Expensify/App/blob/main/STYLING.md)) for all style edits I made
    - [ ] I followed the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs))
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I corroborated the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If I created a new component I verified that a similar component doesn't exist in the codebase

#### PR Reviewer Checklist
- [ ] I verified the PR has a small number of commits behind `main`
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the testing environment is mentioned in the test steps
- [ ] I verified testing steps cover success & fail scenarios (if applicable)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] iOS / native
    - [ ] Android / native
    - [ ] iOS / Safari
    - [ ] Android / Chrome
    - [ ] MacOS / Chrome
    - [ ] MacOS / Desktop
- [ ] I verified there are no console errors related to changes in this PR
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified comments were added when the code was not self explanatory
    - [ ] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [ ] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [ ] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/STYLE.md#jsdocs)) were followed
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components are not impacted by changes in this PR (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [ ] I verified the UI performance was not affected (the performance is the same than `main` branch)
- [ ] If a new component is created I verified that a similar component doesn't exist in the codebase

### QA Steps

1. Open Default Room (#admins and #announce) Settings.
2. Invite and Leave button should be hidden.
3. Open User-Created Room Settings.
4. Invite and Leave button should be visible.

- [x] Verify that no errors appear in the JS console

#### Web

Default Room            |  User-Created Room
:-------------------------:|:-------------------------:
  |  

#### Mobile Web

Default Room            |  User-Created Room
:-------------------------:|:-------------------------:
Simulator Screen Shot - iPhone 13 - 2022-03-07 at 10 46 04 | Simulator Screen Shot - iPhone 13 - 2022-03-07 at 10 45 44

#### Desktop

Default Room            |  User-Created Room
:-------------------------:|:-------------------------:
  |  

#### iOS

Default Room            |  User-Created Room
:-------------------------:|:-------------------------:
Simulator Screen Shot - iPhone 13 - 2022-03-07 at 10 45 14 | Simulator Screen Shot - iPhone 13 - 2022-03-07 at 10 44 28

#### Android

Default Room            |  User-Created Room
:-------------------------:|:-------------------------:
Screenshot_1646628735  |  Screenshot_1646628716

### CODE REVIEW

File: src/pages/ReportDetailsPage.js [hunk 1]
```diff
@@ -82,13 +82,13 @@ class ReportDetailsPage extends Component {
 
         // Chat rooms will allow you to more things than typical chats so they have extra options
         if (ReportUtils.isChatRoom(this.props.report)) {
```
Comment (parasharrajat) [inline src/pages/ReportDetailsPage.js:84]: Let's use early return here.  If not chatroom return.

File: src/pages/ReportDetailsPage.js [hunk 2]
```diff
@@ -82,13 +82,13 @@ class ReportDetailsPage extends Component {
...
-                    icon: Expensicons.Gear,
-                    action: () => { Navigation.navigate(ROUTES.getReportSettingsRoute(props.report.reportID)); },
-                },
-                {
+            this.menuItems.push({
+                translationKey: 'common.settings',
+                icon: Expensicons.Gear,
+                action: () => { Navigation.navigate(ROUTES.getReportSettingsRoute(props.report.reportID)); },
+            });
+            if (!ReportUtils.isDefaultRoom(this.props.report)) {
```
Comment (parasharrajat) [inline src/pages/ReportDetailsPage.js:90]: Same for here. If default room return.

**Review by parasharrajat — CHANGES_REQUESTED** [review]

**Review by parasharrajat — APPROVED** [review]
LGTM. 

cc: @chiragsalian 

#### PR Reviewer Checklist
- [x] I verified the Author pulled `main` before submitting the PR
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the testing environment is mentioned in the test steps
- [x] I verified testing steps cover success & fail scenarios (if applicable)
- [x] I verified tests pass on **all platforms** & I tested again on all platforms
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified there are no console errors related to changes in this PR
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified comments were added when the code was not self explanatory
    - [x] I verified any copy / text shown in the product was added in all `src/languages/*` files (if applicable)
    - [x] I verified proper naming convention for platform-specific files was followed (if applicable)
    - [x] I verified [style guidelines](https://github.com/Expensify/App/blob/main/STYLING.md) were followed
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/PR_REVIEW_GUIDELINES.md)

:ribbon: :eyes: :ribbon:  C+ reviewed

**Review by chiragsalian — APPROVED** [review]
LGTM

**Pull request merged**